Topics/Development_Process/QA_testing.md

```diff
@@ -0,0 +1,51 @@
+
+This paper is likely the greatest paper about software QA, however it is quite long. Here are some of the main definitions and key takeaways:
+- In order to make sure the released software is safe and functions as expected, the concept of software quality was introduced. It is often defined as “the degree of conformance to explicit or implicit requirements and expectations”.
+- The purpose of QA is to set up adequate processes, introducing the standards of quality to prevent the errors and flaws in the product. This is slightly different to testing where the purpose is to detect and solve software errors and flaws.
+- There are different levels of software testing, component/unit testing, integration testing, system testing then acceptance testing.
+
+
+This article gives a tease of what a software QA engineer is like, the different types of SQA engineers and their duties and responsibilities. It is the right article to figure out if you want a career in this field or you are interested in QA.
+[What does a Software Quality assurance engineer do?](https://www.careerexplorer.com/careers/software-quality-assurance-engineer/)
+
+## Why should we test?
+
+Testing is a vital part of the software development life cycle.
+[This very entertaining video](https://youtu.be/oLc9gVM8FBM) shows what events prompted the creation of software testing and how it has changed. The first few minutes of the video is what I would like to call attention to. It highlights the importance of software testing as far back as 1999 with the "Y2K" problem. Every software engineer should know about this.
+
+
+There are many reasons why software testing is important. This blog [7 reasons why software testing is important](https://www.indiumsoftware.com/blog/why-software-testing/) summarizes it into 7 different reasons:
+- Helps in Saving money: Finding a bug, especially when it is found early in the development process reduces costs considerably.
+- Security: Oftentimes software can be vulnerable to attack, or the information of users can be stolen. It is important to avoid this with proper testing.
+- Quality of product: The product should function in a complete manner to ensure effective customer experience.
+- Satisfaction of the Customer: This should be the primary objective of the owner and if the customer is not happy then they will use another product.
+- Enhance the development process: Quality assurance allows for the developer to find a wide array of scenarios and errors.
+- Easy while adding new features: When changing old code it can sometimes break it without the developer knowing. Testing is a great way to give the developer some confidence when adding a new feature.
+- Determining the performance of the software: If your application has a low or reduced perfomance, it will bring a bad reputation in the market. Software testing can help identify these issues before users do.
```

Comment:
```suggestion
- Determining the performance of the software: If your application has a low or reduced performance, it will bring a bad reputation in the market. Software testing can help identify these issues before users do.
```